src/DeterministicNoiseSimulator.cpp

```diff
@@ -578,35 +628,47 @@ void DeterministicNoiseSimulator::generateGate(qc::MatrixDD* pointer_for_matrice
 }
 
 void DeterministicNoiseSimulator::applyDetNoiseSequential(const qc::Targets& targets) {
-    qc::MatrixDD tmp = {};
+    dd::dEdge tmp = {};
     //    qc::MatrixDD ancillary_edge_1 = {};
     qc::MatrixDD idle_operation[4];
 
     // Iterate over qubits and check if the qubit had been used
     for (auto target_qubit: targets) {
         for (auto const& type: gateNoiseTypes) {
-            generateGate(idle_operation, type, target_qubit);
+            double probability;
+            if (type == 'A') {
+                probability = (targets.size() == 1) ? ampDampingProbSingleQubit : ampDampingProbMultiQubit;
+            } else {
+                probability = (targets.size() == 1) ? noiseProbSingleQubit : noiseProbMultiQubit;
+            }
+            generateGate(idle_operation, type, target_qubit, probability);
             tmp.p = nullptr;
             //Apply all noise matrices of the current noise effect
-            for (int m = 0; m < noiseEffects.find(type)->second; m++) {
-                auto tmp0 = dd->multiply(dd->multiply(idle_operation[m], density_root_edge), dd->conjugateTranspose(idle_operation[m]));
+            for (int m = 0; m < gateNoiseEffects.find(type)->second; m++) {
+                auto tmp0 = dd->conjugateTranspose(idle_operation[m]);
+                auto tmp1 = dd->multiply(density_root_edge, reinterpret_cast<dEdge&>(tmp0), 0, false);
+                auto tmp2 = dd->multiply(reinterpret_cast<dEdge&>(idle_operation[m]), tmp1, 0, use_density_matrix_type);
```

Comment:
as mentioned before, this should be encapsulated into a function

Reply:
I created such a function.